### PR TITLE
Specs views

### PR DESCRIPTION
**Pull request by hiromataba**
In this PR we did the following :point_down: 
- Use Capybara to write integration tests for each view in your project.

- Login page:

   - I can see the username and password inputs and the "Submit" button.
   - When I click the submit button without filling in the username and the password, I get a detailed error.
   - When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
   - When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
- User index page:

   - I can see the username of all other users.
   - I can see the profile picture for each user.
   - I can see the number of posts each user has written.
   - When I click on a user, I am redirected to that user's show page.
- user show page:

   - I can see the user's profile picture.
   - I can see the user's username.
   - I can see the number of posts the user has written.
   - I can see the user's bio.
   - I can see the user's first 3 posts.
   - I can see a button that lets me view all of a user's posts.
   - When I click a user's post, it redirects me to that post's show page.
   - When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:

   - I can see the user's profile picture.
   - I can see the user's username.
   - I can see the number of posts the user has written.
   - I can see a post's title.
   - I can see some of the post's body.
   - I can see the first comments on a post.
   - I can see how many comments a post has.
   - I can see how many likes a post has.
   - I can see a section for pagination if there are more posts than fit on the view.
   - When I click on a post, it redirects me to that post's show page.
- Post show page:

   - I can see the post's title.
   - I can see who wrote the post.
   - I can see how many comments it has.
   - I can see how many likes it has.
   - I can see the post body.
   - I can see the username of each commenter.
   - I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/user/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+user_index_spec.rb
```
Comment (Lameck1) [inline spec/features/user/user_index_spec.rb:1]: - [ ] Consider deleting this line, it makes running the tests fail. See the screenshot below 👇🏼:
error

File: spec/features/user/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+user_show_spec.rb
```
Comment (Lameck1) [inline spec/features/user/user_show_spec.rb:1]: - [ ] Consider deleting this one too, the same issue as above. Screenshot attached below 👇🏼:
error_2

File: spec/features/user/user_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      sign_in user
+      visit root_path
+    end
+    it 'should be able see the username of all other users.' do
+      expect(page).to have_content('Maria')
+      expect(page).to have_content('Carlo')
+    end
+    it 'should be able see the number of posts each user has written.' do
+      expect(page).to have_content('Number of posts:3')
+    end
```
Comment (Lameck1) [inline spec/features/user/user_index_spec.rb:19]: - [ ] You've done well writing featured tests for User index page. However the expected number of examples here should be four. Please add examples for the following:

1. Tests that asserts you can see the profile picture for each user.
2. Tests that redirects to the user's show page when you click on a user.

File: spec/features/user/user_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    it 'should see a button.' do
+      expect(page).to have_content('See all posts')
+    end
+    it "should be able see the user's bio." do
+      expect(page).to have_content('This is my bio.')
+    end
+    it "should redirects me to the user's post's index page" do
+      find_link("see_all_#{@user_one.id}").click
+      expect(current_path).to eq(posts_path(@user_one.id))
+    end
```
Comment (Lameck1) [inline spec/features/user/user_show_spec.rb:29]: - [ ] In featured tests for user show page, we also have a number of example not implemented. Please implement them 👇🏼:

1. Tests that asserts you can see the user's profile picture.
2. Tests that asserts you can see the user's first 3 posts.
3. Tests that redirects to the post's show page when you click on a user's post.

File: spec/features/post/post_index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+RSpec.describe 'Post view', type: :feature do
+  include Devise::Test::IntegrationHelpers
+  describe 'Post index' do
+    let(:user) { User.create(name: 'denis', email: 'denis@example.com', password: 'password') }
+    describe 'display post information' do
```
Comment (Lameck1) [inline spec/features/post/post_index_spec.rb:6]: - [ ] Please add the following missing example to featured tests for Post index
1. Tests that asserts you can see the user's profile picture.
2. Tests that redirects to the post's show page when you click on a post.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @hiromataba & @denisdiaconu, 

Well done with writing integration tests for views 👏🏼.
You've done such a splendid job but there are a few tests/edge cases from the requirements that we've missed. We need to add those so that our integration tests can be complete.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
